File: nitpickers/pyql/pyqls/ast/nodes/section.py
from util.ast import ASTNode


class Section(ASTNode):

    def __init__(self, location, name, block):
        super().__init__(location)
        self._name = name
        self._block = block

    @property
    def name(self):
        return self._name

    @property
    def block(self):
        return self._block

    def __repr__(self):
        return "Section at {0}".format(self.location)

    __str__ = __repr__

File: piotr/org.uva.sc.piotr.qls/src/main/java/qls/ast/model/Section.java
package qls.ast.model;

import qls.ast.visitors.ASTNodeVisitor;

import java.util.List;
import java.util.stream.Collectors;

public class Section extends BlockElement {

    private final String name;
    private final List<BlockElement> elements;

    public Section(MetaInformation metaInformation, String name, List<BlockElement> elements) {
        super(metaInformation);
        this.name = name;
        this.elements = elements;
    }

    public String getName() {
        return name;
    }

    public List<BlockElement> getElements() {
        return elements;
    }

    public List<DefaultDefinition> getDefaultDefinitions() {
        return this.elements.stream()
                .filter(item -> item instanceof DefaultDefinition)
                .map(item -> (DefaultDefinition) item)
                .collect(Collectors.toList())
                ;
    }

    @Override
    public <T> T accept(ASTNodeVisitor<T> visitor) {
        return visitor.visit(this);
    }
}
